crates/cxx-qt-lib-headers/include/common.h

```diff
@@ -0,0 +1,35 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#pragma once
+
+namespace rust {
+namespace cxxqtlib1 {
+
+template<typename T>
+void
+drop(T& value)
+{
+  value.~T();
+}
+
+template<typename T>
+T
+initDefault()
+{
+  return T();
+}
+
+template<typename T>
+T
+clone(const T& value)
```

Comment:
This is where we could try getting creative:
How about a `construct` method, that can take any argument?
e.g.:
```
template<typename... Args>
T construct(Args.. args) {
    return T(args...);
}
```

Then even stuff like `qDateInit`, `qdatetimeInitFromDateAndTime` could all boil down to this one template

Reply:
We wouldn't even need the `initDefault` method then, just `construct` without arguments would be equivalent.

Reply:
Done